app/uk/gov/hmrc/tai/service/IncomeService.scala

```diff
@@ -36,21 +37,23 @@ import scala.concurrent.Future
 class IncomeService @Inject()(
   taxAccountService: TaxAccountService,
   employmentService: EmploymentService,
-  taiConnector: TaiConnector) {
+  taiConnector: TaiConnector
+) {
 
   def employmentAmount(nino: Nino, id: Int)(implicit hc: HeaderCarrier, messages: Messages): Future[EmploymentAmount] =
-    for {
-      taxCodeIncomeDetails <- taxAccountService.taxCodeIncomes(nino, TaxYear())
-      employmentDetails    <- employmentService.employment(nino, id)
-    } yield {
-      (taxCodeIncomeDetails, employmentDetails) match {
-        case (TaiSuccessResponseWithPayload(taxCodeIncomes: Seq[TaxCodeIncome]), Some(employment)) =>
-          taxCodeIncomes.find(_.employmentId.contains(id)) match {
-            case Some(taxCodeIncome) => EmploymentAmount(taxCodeIncome, employment)
-            case _                   => throw new RuntimeException(s"Not able to found employment with id $id")
-          }
-        case _ => throw new RuntimeException("Exception while reading employment and tax code details")
-      }
+    (
+      taxAccountService.taxCodeIncomes(nino, TaxYear()),
+      employmentService.employment(nino, id)
+    ) mapN {
+      case (taxCodeIncomeDetails, employmentDetails) =>
+        (taxCodeIncomeDetails, employmentDetails) match {
```

Comment:
I think this can be reduced to
 
```
    (
      taxAccountService.taxCodeIncomes(nino, TaxYear()),
      employmentService.employment(nino, id)
    ) mapN {
          case (TaiSuccessResponseWithPayload(taxCodeIncomes: Seq[TaxCodeIncome]), Some(employment)) =>
            taxCodeIncomes.find(_.employmentId.contains(id)) match {
              case Some(taxCodeIncome) => EmploymentAmount(taxCodeIncome, employment)
              case _                   => throw new RuntimeException(s"Not able to found employment with id $id")
            }
          case _ => throw new RuntimeException("Exception while reading employment and tax code details")
        }
```

cc @sergi-hmrc

Reply:
yup, the match is unnecessary

Reply:
makes sense!